### ISSUE TITLE
Interface of plot function; parameters slipping through.

### ISSUE THREAD

**Issue by rwijtvliet** [geopandas/geopandas#1386]
While fixing [1384](https://github.com/geopandas/geopandas/issues/1384) I noticed the following.

The `mpl` interface accepts both the parameters `linewidth` and `color`, and `linewidths` and `colors`. If the caller of the `GeoDataFrame.plot` method chooses to use the plural parameter names (as they might expect to be able to, just as they can when using `mpl` directly), these are silently passing through, without being subjected to the checks that are in place for `linewidth` and `color`.

Here is an example (using the repo in [this PR](https://github.com/geopandas/geopandas/pull/1385)):

```python
import geopandas as gpd
import shapely.geometry as sg

gdf1 = gpd.GeoDataFrame([{'geometry':sg.MultiLineString([[[0,0],[0,1]],[[0,1],[1,1]]]),
                         'linewidth':8, 'color':'red'},
                        {'geometry':sg.LineString([[0,0],[2,1]]),
                         'linewidth':2, 'color':'blue'}])
gdf1.plot(linewidth=gdf1['linewidth'], color=gdf1['color']) #plots correctly
```
![image](https://user-images.githubusercontent.com/4106013/79672563-91a7d780-81d3-11ea-8464-7390d79d0c05.png)

```python
gdf1.plot(linewidths=gdf1['linewidth'], colors=gdf1['color']) # plots incorrectly
```
![image](https://user-images.githubusercontent.com/4106013/79672572-9b313f80-81d3-11ea-980c-1770d536ad2a.png)

I'm not sure if we should do something about it; plotting functions have so many parameters... It's not too hard to also check and expand the plural parameter names, but I'd like to discuss first.

**Comment by martinfleis** [comment]
This is good catch. I didn't know Matplotlib supports that. It is a bit corner case, but if we make suggested changed in #1385, that should resolve this as well.

**Comment by martinfleis** [comment]
Fixed by #1385.
